src/Snackbar/Snackbar.js

```diff
@@ -0,0 +1,229 @@
+// @flow weak
+
+import React, { PropTypes, Component } from 'react';
+import { createStyleSheet } from 'jss-theme-reactor';
+import customPropTypes from 'material-ui/utils/customPropTypes';
+import SnackbarContent from './SnackbarContent';
+import ClickAwayListener from '../internal/ClickAwayListener';
+
+export const styleSheet = createStyleSheet('Snackbar', (theme) => {
+  const { zIndex } = theme;
+  return {
+    root: {
+      position: 'fixed',
+      left: '50%',
+      display: 'flex',
+      bottom: 0,
+      zIndex: zIndex.snackbar,
+      transition: `transform 400ms 0ms cubic-bezier(0.23, 1, 0.32, 1),
+       visibility 400ms 0ms cubic-bezier(0.23, 1, 0.32, 1)`,
+    },
+    '@media(max-width:759px )': {
+      root: {
+        minWidth: 'inherit',
+        maxWidth: 'inherit',
+        width: '100%',
+        display: 'block',
+      },
+    },
+  };
+});
+
+export default class Snackbar extends Component {
+
+  static propTypes = {
+    /**
+     * The label for the action on the snackbar.
+     */
+    action: PropTypes.node,
+    /**
+     * The number of milliseconds to wait before automatically dismissing.
+     * If no value is specified the snackbar will dismiss normally.
+     * If a value is provided the snackbar can still be dismissed normally.
+     * If a snackbar is dismissed before the timer expires, the timer will be cleared.
+     */
+    autoHideDuration: PropTypes.number,
+    /**
+     * The css class name of the root element.
+     */
+    className: PropTypes.string,
+    /**
+     * The message to be displayed.
+     *
+     * (Note: If the message is an element or array,
+     *  and the `Snackbar` may re-render while it is still open,
+     *  ensure that the same object remains as the `message` property
+     *  if you want to avoid the `Snackbar` hiding and
+     *  showing again)
+     */
+    message: PropTypes.node.isRequired,
+    /**
+     * Fired when the `Snackbar` is requested to be closed by a
+     * click outside the `Snackbar`, or after the
+     * `autoHideDuration` timer expires.
+     *
+     * Typically `onRequestClose` is used to set state in the parent component,
+     * which is used to control the `Snackbar`
+     * `open` prop.
+     *
+     * The `reason` parameter can optionally be used to control the response to `onRequestClose`,
+     * for example ignoring `clickaway`.
+     *
+     * @param {string} reason Can be:`"timeout"` (`autoHideDuration` expired) or: `"clickaway"`
+     */
+    onRequestClose: PropTypes.func,
+    /**
+     * Controls whether the `Snackbar` is opened or not.
+     */
+    open: PropTypes.bool.isRequired,
+    /**
+     * Override the inline-styles of the root element.
+     */
+    style: PropTypes.object,
+
+  };
+
+  state = {
```

Comment:
Here is how I the component. I would have 3 modules to handle 3 concerns:
1. **The state**. A Higher order Component seems suited, but I'm fine with any alternative. It's going to handle the open/close logic when the message changes. I don't have much idea of the name. We could call it `<SnackbarStateful />` or `withStatefulLogic`.
2. **The positioning** (`Snackbar`). It's going to handle the fixed positioning as well as the ClickAway callback and the autohide logic. By the way, the autohide logic could be extended in order to be pause when the user mouse is over the snackbar.
3. **The UI** (`SnackbarContent`). We can add visual regression test against that one.